src/App.tsx

```
@@ -1,61 +1,141 @@
+import React, { useState } from 'react';
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
 
 export const App = () => {
+  const userMap = new Map(usersFromServer.map(u => [u.id, u]));
+
+  const initialTodos: Todo[] = todosFromServer
+    .map(todo => {
+      const user = userMap.get(todo.userId);
+
+      if (!user) {
+        return null;
+      }
+
+      return {
+        ...todo,
+        user,
+      };
+    })
+    .filter((todo): todo is Todo => todo !== null);
+
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState<number | ''>('');
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleChange = (
+    changeEvent: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const cleaned = changeEvent.target.value.replace(
+      /[^a-zA-Z\u0400-\u04FF0-9\s]/g,
+      '',
+    );
+
+    setTitle(cleaned);
+    if (titleError && cleaned.trim() !== '') {
+      setTitleError(false);
+    }
+  };
```

Validation hide behavior does not match the task description requirement. The description states: "hide the message immediately after any change of the field with an error;" but the title change handler only clears the error when the cleaned value becomes non-empty (`if (titleError && cleaned.trim() !== '') { setTitleError(false); }`). Instead, if `titleError` is true then hide the error as soon as the input changes (i.e., clear the error immediately on any change after submit). This ensures the error disappears immediately after the user's first change as required. See description for the exact phrase.